osquery/remote/http_client.cpp

```diff
@@ -203,8 +203,12 @@ void Client::createConnection() {
 
 void Client::encryptConnection() {
   boost::asio::ssl::context ctx{boost::asio::ssl::context::sslv23};
+  const auto remote_hostname = client_options_.remote_hostname_->c_str();
 
   if (client_options_.always_verify_peer_) {
+    ssl_sock_->set_verify_callback(
+        boost::asio::ssl::rfc2818_verification(remote_hostname));
+
```

Comment:
This has to be moved after here https://github.com/osquery/osquery/pull/6197/files#diff-2e3d7b31ff943d22ef0475f6645e440aR246, otherwise `ssl_sock_` is not initialized.

Reply:
I originally planned to close this PR, as we could just add this single line inside the base PR. Feel free to remove the `do-not-merge` tag if you want to update & merge this one!